### PR TITLE
Unbalanced quotes are malformed expressions

### LINKED ISSUE #6: Fix parser ambiguities...
These will be added with tests in #4

### PR DESCRIPTION
**Pull request by stanistan**
Fixes #6

### CODE REVIEW

File: src/grammar.js [hunk 1]
```diff
@@ -33,12 +35,12 @@ const grammar = ohm.grammar(`
     hole
       = "_"
 
-    word
-      = (~special any)+  -- nonspace
-      | space+           -- space
+    ws
```
Comment (jedahan) [inline src/grammar.js]: maybe call this `spaces`?

File: src/grammar.js [hunk 2]
```diff
@@ -53,9 +55,8 @@ const grammar = ohm.grammar(`
     string
       = "\\"" (~"\\"" ~"\\n" any)* "\\""
 
-    special
-      = id | value | variable | wildcard | hole | space
-
+    word_char = ~("#" | "\\"" | "$" | "_" | space | digit | ".") any
```
Comment (jedahan) [inline src/grammar.js]: By getting rid of special, we need to keep two lists of definitions of characters in sync. Not a huge deal, but is there some ohm magic we could use to keep that connection? something closer to
```
word
  = ~special any
  | (~special any)+ "."+
```
This is unfamiliar territory for me so I'm not sure if my question makes sense

File: src/grammar.js [hunk 3]
```diff
@@ -7,23 +7,25 @@ const grammar = ohm.grammar(`
...
+      | word
+      | ws
 
     value
-      = keyword<"true">   -- true
-      | keyword<"false">  -- false
-      | keyword<"null">   -- null
+      = "true"   -- true
+      | "false"  -- false
+      | "null"   -- null
```
Comment (jedahan) [inline src/grammar.js]: I don't think we need the comments here anymore.

I like the cleanup, but are we losing anything in removing the distinction between
"things prefixed with symbols which clearly indicate that there is some special meaning"
vs
"alphanumeric works we just decided have special meaning and might be confusing that they parse as a value?"

For first time contributors, are they losing debugging information when their sentence goes wrong? Right now I don't think so, so this should be fine.

**Review by jedahan — APPROVED** [review]
Only one 'real' question, re word_char and word_pattern. Otherwise, more passing tests the better. I got bit with an unclosed quote today :)

File: src/grammar.js (hunk 3 above)
Comment (stanistan) [reply src/grammar.js]: > I don't think we need the comments here anymore.

The comments do dispatch within the semantics so we still need that (if I understand what ohm do).

---

> For first time contributors, are they losing debugging information when their sentence goes wrong? Right now I don't think so, so this should be fine.

We don't really have super descriptive error messages right now in the parser either, I think we can get them out of the parser generator using [comments](https://github.com/harc/ohm/blob/master/doc/syntax-reference.md#rule-descriptions)?

File: src/grammar.js (hunk 2 above)
Comment (jedahan) [reply src/grammar.js]: maybe `symbol` or `mark`

File: src/grammar.js (hunk 3 above)
Comment (jedahan) [reply src/grammar.js]: oh dang, definitely would love comments

and misread -- as lua comment

Comment (stanistan) [reply src/grammar.js]: Was playing around with `ohm` and hit this https://github.com/harc/ohm/pull/218 🤷‍♀️

**Pull request merged**